### DISCUSSION TITLE
Use selenium driver to get webpage content ?

### DISCUSSION THREAD

**Discussion by ikus060** [FreshRSS/FreshRSS#6628] [Ideas]
Have you ever consider making use of selenium driver to fetch the web page of web site ? 

Nowaday, many website rely on javascript execution to display page content which make it close to impossible to use the XPath + HTML web scrapper provided by FreshRss. Selenium Driver may bypass this complexity by creating an instance of a web browser in backend and query the webpage. The javascript could then render the content of the page. After that FreshRss could run the Xpath ?

I have some experience with selenium driver with python. I see it's might work for php too.

This technique should allow web scrapping of website like twitter.

**Comment by lukasMega** [comment]
Selenium is good, however I think the best times are past. Very good (and maybe better) alternatives are: • Puppeteer (https://pptr.dev/) and/or • Playwright (https://playwright.dev/) 
- these alternatives also uses real instance of browser (or headless version)
- there are also lot of useful plugins, for example re-captcha https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-recaptcha

But I don't think that advanced web scrapping should be part of FreshRSS application. But it's good candidate of a plugin/extension.

**Reply by Alkarex** [reply]
Indeed, and extensions welcome
